runner/ansible/README.md

```diff
@@ -22,3 +41,84 @@ These are the fields needed by Trento:
 is that the labels are used for control purpose (select all the checks with this label e.g.), and the groups are used just for visual purposes
 - `description`: A longer description about the test purpose. It can be written using markdown.
 - `implementation`: Usually the task `main.yml` content
+
+## Check files
+
+The check files contain the actual task which the runner will execute. An
+example is available at [tasks/main.yml](roles/checks/1.1.1/tasks/main.yml).
+These files are written like a normal ansible task, but:
+
+- The tasks are of `read only` nature, meaning that they are meant to check for things
+  rather than something beeing executed.
+- Their output is either `true` (check passed) or `false` (check failed).
+  This output will then be passed to the `post-results` block with the `status` 
+  field containing the result of the check. 
+
+```
+name: "{{ id }}.check"
+```
+
+```  
+check_mode: no
+register: config_updated
+changed_when: config_updated.stdout != expected[id]
+```
+```
+- block:
+    - import_role:
+        name: post-results
+  when:
+    - ansible_check_mode
+  vars:
+    status: "{{ config_updated is not changed }}"
+``` 
+
+## Examples
+
+As an example (we will use `dummy`), lets add a test which simply creates a simple file in /tmp.
+Fist we want to define the metadata in `runner/ansible/roles/checks/dummy/defaults/main.yml`:
+
+```
+---
+
+id: dummy
+name: Check os-release file
+group: Testing
+labels: generic
+description: |
+  Checks for the os-release file in /etc/os-release
+remediation: |
+  ## Remediation
+  This check is for exemplary purposes
+
+  ## References
+  Place for references
+implementation: "{{ lookup('file', 'roles/checks/'+id+'/tasks/main.yml') }}"
+
+# Test data
+key_name: token
+``` 
+
+Then the actual task `runner/ansible/roles/checks/3.0.0/tast/main.yml`:
+
+```
+---
+
+- name: "{{ id }}.check"
+  stat:
+      path: "/etc/os-release"
+  check_mode: no
+  register: config_updated
+  changed_when: config_updated.stdout != expected[id]
```

Comment:
This `expected[id]` might lead to a confusion, as it is not mentioned anywhere.
As a summary, the `expected` values are in the `vars` folder, and they might be updated by the user (from the web interface in the future)